src/DynamoCore/Linting/LinterManager.cs

```diff
@@ -0,0 +1,124 @@
+        private LinterExtensionDescriptor activeLinter;
+        #endregion
+
+        #region Public properties
+
+        /// <summary>
+        /// Available linters
+        /// </summary>
+        public HashSet<LinterExtensionDescriptor> AvailableLinters { get; internal set; }
+
+        /// <summary>
+        /// Results from evaluated rules
+        /// </summary>
+        public ObservableCollection<IRuleEvaluationResult> RuleEvaluationResults { get; set; }
+
+        /// <summary>
+        /// The LinterDescripter that is currently set as active
+        /// </summary>
+        public LinterExtensionDescriptor ActiveLinter
+        {
+            get => activeLinter;
+            set
+            {
+                if (activeLinter == value)
+                    return;
+
+                activeLinter = value;
+
+                var linterExt = GetLinterExtension(activeLinter);
+                if (linterExt is null)
+                    return;
+
+                linterExt.Activate();
+            }
+        }
+
+        #endregion
+
+        public LinterManager(DynamoModel dynamoModel)
+        {
+            this.dynamoModel = dynamoModel;
+            AvailableLinters = new HashSet<LinterExtensionDescriptor>();
+            RuleEvaluationResults = new ObservableCollection<IRuleEvaluationResult>();
+
+            SubscribeLinterEvents();
+        }
+
+        /// <summary>
+        /// Checks if the uniqueId equals the id of the Active linter descriptor
+        /// </summary>
+        /// <param name="uniqueId"></param>
+        /// <returns></returns>
+        internal bool IsExtensionActive(string uniqueId)
+        {
+            return ActiveLinter?.Id == uniqueId;
+        }
+
+        #region Private methods
+        private void SubscribeLinterEvents()
+        {
+            LinterExtensionBase.LinterExtensionReady += OnLinterExtensionReady;
+            LinterRule.RuleEvaluated += OnRuleEvaluated;
+
+        }
+
+        private void OnLinterExtensionReady(LinterExtensionDescriptor extensionDescriptor)
+        {
+            if (AvailableLinters.Contains(extensionDescriptor))
+                return;
+
+            AvailableLinters.Add(extensionDescriptor);
+        }
+
+        private void OnRuleEvaluated(IRuleEvaluationResult result)
+        {
+            if (result is null)
+                return;
+
+            if (result.Status == RuleEvaluationStatusEnum.Passed)
+            {
+                if (!RuleEvaluationResults.Contains(result))
+                    return;
+                RuleEvaluationResults.Remove(result);
+            }
+
+            else
+            {
+                if (RuleEvaluationResults.Contains(result))
+                    return;
+                RuleEvaluationResults.Add(result);
+            }
+        }
+
+        private LinterExtensionBase GetLinterExtension(LinterExtensionDescriptor activeLinter)
+        {
+            return this.dynamoModel.ExtensionManager.
```

Comment:
why pass make this type depend on DynamoModel when all you use is the ExtensionManager?